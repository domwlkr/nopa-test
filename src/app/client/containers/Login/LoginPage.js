import React from 'react';
import {connect} from 'react-redux';
import { Layout, LoginForm } from '../../components';

class LoginPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      formData: {
        surname: '',
        sortCode: '',
        accountNumber: '',
        passCode: '',
        memorableWord: ''  
      },
      errors: {
        surname: '',
        sortCode: '',
        accountNumber: '',
        passCode: '',
        memorableWord: ''
      }
    };
  }

  handleChange(event) {
    let newState = this.state;

    newState.formData[event.target.name] = event.target.value;

    this.setState(newState);
  }

  handleSubmit(event) {
    event.preventDefault();
    this.validate();
  }

  validate() {
    let newState = this.state;
    let errors = {};

    if (this.state.formData.surname.length < 1) {
      errors.surname = 'Please enter a surname';
    }

    if (this.state.formData.sortCode.length < 1) {
      errors.sortCode = 'Please enter a sort code';
    }

    if (this.state.formData.accountNumber.length < 1) {
      errors.accountNumber = 'Please enter an account number';
    }

    if (this.state.formData.passCode.length < 1) {
      errors.passCode = 'Please enter a pass code';
    }

    if (this.state.formData.memorableWord.length < 1) {
      errors.memorableWord = 'Please enter a memorable word';
    }

    newState.errors = errors;

    this.setState(newState);
  }

  render() {
    return (
      <Layout title="Login">
        <div className="main-content">
          <h1>Which bank does this account belong to?</h1>
          <p>Track all of your payments by connecting as many bank accounts as you'd<br />
            like to your Nopa account and get updates on your balance instantly. Plus it's free.</p>

            <LoginForm
              handleChange={(event) => this.handleChange(event)}
              handleSubmit={(event) => this.handleSubmit(event)}
              errors={this.state.errors} />
        </div>
      </Layout>
    );  
  }
  
}

export default LoginPage;
