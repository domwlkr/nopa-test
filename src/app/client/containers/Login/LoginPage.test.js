import React from 'react';
import { LoginPage } from './LoginPage';
import { Provider } from 'react-redux';
import configureStore from '../../redux/store/configureStore';
import renderer from 'react-test-renderer';

const store = configureStore();

test('Renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}><LoginPage /></Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
