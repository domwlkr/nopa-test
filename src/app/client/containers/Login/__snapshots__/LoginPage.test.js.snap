// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section
  className={undefined}
>
  <div
    className="header"
  >
    <img
      className="logo"
      src="test-file-stub"
    />
    <div
      className="login"
    >
      <a
        className="button"
        onClick={[Function]}
        style={Object {}}
      >
        Log In
      </a>
    </div>
  </div>
  <div
    className="layout"
  >
    <div
      className="main-content"
    >
      <h1>
        Which bank does this account belong to?
      </h1>
      <p>
        Track all of your payments by connecting as many bank accounts as you'd
        <br />
        like to your Nopa account and get updates on your balance instantly. Plus it's free.
      </p>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-input"
        >
          <label
            htmlFor="surname"
          >
            Surname
          </label>
          <input
            data-error={undefined}
            id="surname"
            name="surname"
            onChange={[Function]}
            placeholder="Surname"
            type="text"
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="sortCode"
          >
            Sort code
          </label>
          <input
            data-error={undefined}
            id="sortCode"
            name="sortCode"
            onChange={[Function]}
            placeholder="Sort code"
            type="text"
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="accountNumber"
          >
            Account number
          </label>
          <input
            data-error={undefined}
            id="accountNumber"
            name="accountNumber"
            onChange={[Function]}
            placeholder="Account number"
            type="text"
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="passCode"
          >
            Passode
          </label>
          <input
            data-error={undefined}
            id="passCode"
            name="passCode"
            onChange={[Function]}
            placeholder="Passode"
            type="text"
          />
          <div
            className="errors"
          />
        </div>
        <div
          className="form-input"
        >
          <label
            htmlFor="memorableWord"
          >
            Memorable word
          </label>
          <input
            data-error={undefined}
            id="memorableWord"
            name="memorableWord"
            onChange={[Function]}
            placeholder="Memorable word"
            type="text"
          />
          <div
            className="errors"
          />
        </div>
        <input
          className="button"
          type="submit"
          value="Login & connect"
        />
      </form>
    </div>
  </div>
  <section
    className="partners"
  >
    <div>
      <div>
        <span>
          Our partners:
        </span>
      </div>
      <div>
        <div
          className="partner"
        >
          <img
            alt="Airbnb"
            src="test-file-stub"
          />
        </div>
        <div
          className="partner"
        >
          <img
            alt="Metro"
            src="test-file-stub"
          />
        </div>
        <div
          className="partner"
        >
          <img
            alt="Pariti"
            src="test-file-stub"
          />
        </div>
        <div
          className="partner"
        >
          <img
            alt="Unshackled"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </section>
  <footer
    className="footer"
  >
    <div>
      <p>
        © Nopa Limited 2017 All rights reserved. ‘Nopa’ and the Nopa logo are trade marks of Nopa Limited. Nopa is a member of CIFAS - the UK's leading anti-fraud association, and we are registered with the Office of the Information Commissioner (No. Z879078).
      </p>
      <p>
        Nopa Limited is incorporated in England & Wales (registration number 05197592), with its registered office at 1st Floor, Cottons Centre, Tooley Street, London, SE1 2QG. Nopa Limited is authorised and regulated by the Financial Conduct Authority, and entered on the Financial Services Register under firm registration number 563134.
      </p>
    </div>
  </footer>
</section>
`;
